unitTests/ui/QuerySuggestAddonTest.ts

```diff
@@ -188,5 +188,45 @@ export function QuerySuggestAddonTest() {
         done();
       });
     });
+
+    describe('when the characterThresholdForSuggestions option is set to a value', () => {
+      let magicBoxText;
+      let spyOnEvent;
+      beforeEach(() => {
+        simulateFakeSuggestions({ completions: [{ executableConfidence: 1, expression: 'foo', highlighted: 'foo', score: 1 }] });
+        magicBoxText = jasmine.createSpy('magicBoxText');
+        spyOnEvent = jasmine.createSpy('spyOnEvent');
+        omnibox.cmp.options.characterThresholdForSuggestions = 3;
+        omnibox.cmp.magicBox.getText = magicBoxText;
+        $$(omnibox.env.root).on(OmniboxEvents.querySuggestSuccess, () => spyOnEvent());
+      });
+
+      it('when the magic box text length is equals to the minimumNumberOfCharactersForSuggestions option', async done => {
+        magicBoxText.and.returnValue('foo');
+        querySuggest = new QuerySuggestAddon(omnibox.cmp);
+        await querySuggest.getSuggestion();
+        expect(magicBoxText).toHaveBeenCalled();
+        expect(spyOnEvent).toHaveBeenCalled();
+        done();
+      });
+
+      it('when the magic box text length is greater than the minimumNumberOfCharactersForSuggestions option', async done => {
+        magicBoxText.and.returnValue('fooo');
+        querySuggest = new QuerySuggestAddon(omnibox.cmp);
+        await querySuggest.getSuggestion();
+        expect(magicBoxText).toHaveBeenCalled();
+        expect(spyOnEvent).toHaveBeenCalled();
+        done();
+      });
+
+      it('when the magic box text length is less than the minimumNumberOfCharactersForSuggestions option', async done => {
```

Comment:
We also provide the expected result in the descriptions
Here for example:
> when the magic box text length is less than the minimumNumberOfCharactersForSuggestions option
> should not trigger query suggestion

Reply:
Ok will do for all the test